src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -8199,6 +8331,25 @@ void Unit::EnergizeBySpell(Unit* victim, uint32 spellId, int32 damage, Powers po
     SendEnergizeSpellLog(victim, spellId, damage, powerType);
 }
 
+uint32 Unit::CalculateMultistrikeAmount(Unit* victim, int32 baseAmount)
+{
+    // Player totems benefit from player auras
+    if (IsTotem() && GetOwnerGUID().IsPlayer() && GetOwner())
+        return GetOwner()->CalculateMultistrikeAmount(victim, baseAmount);
+
+    // Hack for Storm, Fire and Earth elementals
+    if ((GetEntry() == 77936 || GetEntry() == 15352 || GetEntry() == 15438) && GetOwnerGUID().IsPlayer() && GetOwner())
```

Comment:
Plus if you look in Unit::GetUnitSpellCriticalChance, a similar hack check is used for elementals.